csharp/ql/lib/semmle/code/csharp/dataflow/internal/TaintTrackingPrivate.qll

```diff
@@ -169,3 +169,43 @@ private module Cached {
 }
 
 import Cached
+import SpeculativeTaintFlow
+
+private module SpeculativeTaintFlow {
+  private import semmle.code.csharp.dataflow.internal.ExternalFlow as ExternalFlow
+  private import semmle.code.csharp.dataflow.internal.FlowSummaryImpl as Impl
+
+  private predicate hasTarget(Call call) {
+    exists(Impl::Public::SummarizedCallable sc | sc.getACall() = call)
+    or
+    exists(Impl::Public::NeutralSummaryCallable nc | nc.getACall() = call)
+    or
+    call.getTarget().getUnboundDeclaration() instanceof ExternalFlow::SinkCallable
+    or
+    exists(FlowSummaryImpl::Public::NeutralSinkCallable sc | sc.getACall() = call)
```

Comment:
nit: A sink callable could also have flow through (which is just not modelled as a summary or summary neutral) - so maybe such callables should still be taken into consideration for speculative flow?

Reply:
True. Consider this a starting point that can be further tweaked, and this point can certainly be discussed. My initial take was that neutrals would have had some consideration and hence that they would have had a model added if applicable.